src/EnergyPlus/VentilatedSlab.cc

```diff
@@ -752,261 +698,252 @@ namespace VentilatedSlab {
                                                                                        ObjectIsNotParent);
 
             // Fan information:
-            state.dataVentilatedSlab->VentSlab(Item).FanName = state.dataIPShortCut->cAlphaArgs(25);
+            ventSlab.FanName = state.dataIPShortCut->cAlphaArgs(25);
 
-            if (HVACFan::checkIfFanNameIsAFanSystem(state, state.dataVentilatedSlab->VentSlab(Item).FanName)) {
-                state.dataVentilatedSlab->VentSlab(Item).FanType_Num = DataHVACGlobals::FanType_SystemModelObject;
-                state.dataHVACFan->fanObjs.emplace_back(new HVACFan::FanSystem(state, state.dataVentilatedSlab->VentSlab(Item).FanName));
-                state.dataVentilatedSlab->VentSlab(Item).Fan_Index =
-                    HVACFan::getFanObjectVectorIndex(state, state.dataVentilatedSlab->VentSlab(Item).FanName);
+            if (HVACFan::checkIfFanNameIsAFanSystem(state, ventSlab.FanName)) {
+                ventSlab.FanType_Num = DataHVACGlobals::FanType_SystemModelObject;
+                state.dataHVACFan->fanObjs.emplace_back(new HVACFan::FanSystem(state, ventSlab.FanName));
+                ventSlab.Fan_Index =
+                    HVACFan::getFanObjectVectorIndex(state, ventSlab.FanName);
             } else {
                 bool isNotOkay(false);
-                ValidateComponent(state, "FAN:CONSTANTVOLUME", state.dataVentilatedSlab->VentSlab(Item).FanName, isNotOkay, "GetPIUs");
+                ValidateComponent(state, "FAN:CONSTANTVOLUME", ventSlab.FanName, isNotOkay, "GetPIUs");
                 if (isNotOkay) {
-                    ShowContinueError(state, "In " + CurrentModuleObject + " = " + state.dataVentilatedSlab->VentSlab(Item).Name);
+                    ShowContinueError(state, "In " + CurrentModuleObject + " = " + ventSlab.Name);
                     ErrorsFound = true;
                 }
-                state.dataVentilatedSlab->VentSlab(Item).FanType_Num = DataHVACGlobals::FanType_SimpleConstVolume;
+                ventSlab.FanType_Num = DataHVACGlobals::FanType_SimpleConstVolume;
             }
 
-            if (state.dataVentilatedSlab->VentSlab(Item).OAControlType == FixedOAControl) {
-                state.dataVentilatedSlab->VentSlab(Item).OutAirVolFlow = state.dataVentilatedSlab->VentSlab(Item).MinOutAirVolFlow;
-                state.dataVentilatedSlab->VentSlab(Item).MaxOASchedName = state.dataVentilatedSlab->VentSlab(Item).MinOASchedName;
-                state.dataVentilatedSlab->VentSlab(Item).MaxOASchedPtr =
-                    GetScheduleIndex(state, state.dataVentilatedSlab->VentSlab(Item).MinOASchedName);
+            if (ventSlab.outsideAirControlType == OutsideAirControlType::FixedOAControl) {
+                ventSlab.OutAirVolFlow = ventSlab.MinOutAirVolFlow;
+                ventSlab.MaxOASchedName = ventSlab.MinOASchedName;
+                ventSlab.MaxOASchedPtr =
+                    GetScheduleIndex(state, ventSlab.MinOASchedName);
```

Comment:
ventSlab.MaxOASchedPtr = ventSlab.MinOASchedPtr (when you delete the SchedName vars)